src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,470 @@
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+roachprod ssh $CLUSTER:1 lsblk
+NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+loop0     7:0    0  63.9M  1 loop /snap/core20/2182
+loop1     7:1    0 354.3M  1 loop /snap/google-cloud-cli/223
+loop2     7:2    0    87M  1 loop /snap/lxd/27428
+loop3     7:3    0  39.1M  1 loop /snap/snapd/21184
+sda       8:0    0    10G  0 disk
+├─sda1    8:1    0   9.9G  0 part /
+├─sda14   8:14   0     4M  0 part
+└─sda15   8:15   0   106M  0 part /boot/efi
+nvme0n1 259:0    0   375G  0 disk /mnt/data1
+nvme0n2 259:1    0   375G  0 disk /mnt/data2
+nvme0n3 259:2    0   375G  0 disk /mnt/data3
+nvme0n4 259:3    0   375G  0 disk /mnt/data4
+~~~
+
+Next, we will create scripts for stalling and unstalling the disk on this node. The script will add a line in `/sys/fs/cgroup/system.slice/cockroach-system.service/io.max` to change the [cgroup](https://en.wikipedia.org/wiki/Cgroups) settings "read bytes per second" (`rbps`) and "write bytes per second" (`wbps`) of the disk to a very low value for stalling, and a max value for unstalling.
+
+Create a shell script that will invoke the commands to stall and unstall alternately. We will call this script `wal-flip.sh`. Create `wal-flip.sh` with the content shown below:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+# wal-flip.sh
+# We are alternatively running stall and unstall at increasing periods of 5s up to 40s
+while (true)
+do
+    date
+    echo "Stalling"
+    sudo echo '259:0 rbps=4 wbps=4' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 5
+    date
+    echo "Unstalling"
+    sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 10
+    date
+    echo "Stalling"
+    sudo echo '259:0 rbps=4 wbps=4' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 15
+    date
+    echo "Unstalling"
+    sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 20
+    date
+    echo "Stalling"
+    sudo echo '259:0 rbps=4 wbps=4' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 40
+    echo "UnStalling"
+    sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 600
+    exit
+done
+# eof
+~~~
+
+Next, make the script file executable:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+chmod 777 wal-flip.sh
+~~~
+
+Optionally, you can run a [workload]({% link {{ page.version.version }}/cockroach-workload.md %}) at this point, if you want to see your workload continue during WAL failover. However, the presence of the workload does not have any bearing on the operation of WAL failover.
+
+### 2. WAL Failover during transient disk stalls
+
+#### Cause transient disk stall
+
+Before triggering a WAL Failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that we can observe WAL Failover metrics.
```

Comment:
```suggestion
Before triggering a WAL failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that you can observe WAL failover metrics.
```

Reply:
✅